saleor/graphql/checkout/mutations.py

```diff
@@ -1379,7 +1402,13 @@ def perform_on_shipping_method(
             ),
         )
 
-        delivery_method = convert_to_shipping_method_data(shipping_method)
+        delivery_method = convert_to_shipping_method_data(
+            shipping_method,
+            shipping_models.ShippingMethodChannelListing.objects.filter(
+                shipping_method=shipping_method,
+                channel=checkout_info.channel,
+            ).get(),
```

Comment:
Wouldn't be better to have here `.first()`? I don't see any try/catch here

Reply:
Would you prefer to set shipping_method to None in `fetch_checkout_info` when the listing channel no longer exists? We shouldn't expose this "integrity error" to the API consumers by making the field nullable